torch/optim/early_stopping.py

```diff
@@ -0,0 +1,51 @@
+class EarlyStoppingCriterion(object):
+    """
+    Arguments:
+        patience (int): The maximum number of epochs with no improvement before early stopping should take place
+        mode (str, can only be 'max' or 'min'): To take the maximum or minimum of the score for optimization
+        min_delta (float, optional): Minimum change in the score to qualify as an improvement (default: 0.0)
+    """
+
+    def __init__(self, patience, mode, min_delta=0.0):
+        assert patience >= 0
+        assert mode in {'min', 'max'}
+        assert min_delta >= 0.0
+        self.patience = patience
+        self.mode = mode
+        self.min_delta = min_delta
+
+        self._count = 0
+        self._best_score = None
+        self.is_improved = None
+
+    def step(self, cur_score):
+        """
+        Checks if training should be continued given the current score.
+
+        Arguments:
+            cur_score (float): the current score
+
+        Output:
+            bool: if training should be continued
```

Comment:
I think this is kind of counter intuitive:
```py
early_stopping = EarlyStoppingCriterion()
# ...
if early_stopping:
    # continue?
```

Reply:
Actually, it will be like 
```
early_stopping = EarlyStoppingCriterion(...)
if early_stopping.step(val_score):
    # continue
```

We have `step` function in both `optimizer` and `lr_scheduler` where it's used to make an optimization step. So I assumed the consistency here and made the returned bool reflect the successfulness of the optimization step.